diff --git a/index.d.ts b/index.d.ts
index ac4713d26193656dfb240e59c9499f22e3770521..0c2be3ddc457cbaac7d2f8ea3b0d70dcc60f7612 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -274,7 +274,7 @@ declare namespace __WebpackModuleApi {
 
 interface NodeRequire extends NodeJS.Require {}
 
-declare var require: NodeRequire;
+declare var require: NodeJS.Require;
 
 /**
  * The resource query of the current module.
