diff --git a/dist/gatsby-node.d.ts b/dist/gatsby-node.d.ts
index c49c228750a6b476b7bd24d65b515fccc37ccffa..e1c4c1545e3238d064ea0acbf07d14bbbb944e7b 100644
--- a/dist/gatsby-node.d.ts
+++ b/dist/gatsby-node.d.ts
@@ -14,7 +14,7 @@ export declare const resolvableExtensions: GatsbyNode["resolvableExtensions"];
  */
 export declare const preprocessSource: GatsbyNode["preprocessSource"];
 export declare const createSchemaCustomization: GatsbyNode["createSchemaCustomization"];
-export declare const unstable_shouldOnCreateNode: GatsbyNode["unstable_shouldOnCreateNode"];
+export declare const shouldOnCreateNode: GatsbyNode["shouldOnCreateNode"];
 /**
  * Create Mdx nodes from MDX files.
  */
diff --git a/dist/gatsby-node.js b/dist/gatsby-node.js
index 64e406aefb86e9821aedb4e4d020092f3fad587d..538216335caa9d62226e2982af1b3066cfd39ecd 100644
--- a/dist/gatsby-node.js
+++ b/dist/gatsby-node.js
@@ -5,7 +5,7 @@ var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefau
 Object.defineProperty(exports, "__esModule", {
   value: true
 });
-exports.unstable_shouldOnCreateNode = exports.resolvableExtensions = exports.preprocessSource = exports.pluginOptionsSchema = exports.onPluginInit = exports.onCreateWebpackConfig = exports.onCreatePage = exports.onCreateNode = exports.createSchemaCustomization = void 0;
+exports.shouldOnCreateNode = exports.resolvableExtensions = exports.preprocessSource = exports.pluginOptionsSchema = exports.onPluginInit = exports.onCreateWebpackConfig = exports.onCreatePage = exports.onCreateNode = exports.createSchemaCustomization = void 0;
 
 var _path = _interopRequireDefault(require("path"));
 
@@ -279,7 +279,7 @@ const createSchemaCustomization = async ({
 
 exports.createSchemaCustomization = createSchemaCustomization;
 
-const unstable_shouldOnCreateNode = ({
+const shouldOnCreateNode = ({
   node
 }, pluginOptions) => {
   const {
@@ -292,7 +292,7 @@ const unstable_shouldOnCreateNode = ({
  */
 
 
-exports.unstable_shouldOnCreateNode = unstable_shouldOnCreateNode;
+exports.shouldOnCreateNode = shouldOnCreateNode;
 
 const onCreateNode = async ({
   node,
diff --git a/gatsby/on-create-node.js b/gatsby/on-create-node.js
index 0f7488e8c5686785aa27264beb1af5342bab3ae7..b70b570b3f5c42ab0e00bd51831af01cedd6e384 100644
--- a/gatsby/on-create-node.js
+++ b/gatsby/on-create-node.js
@@ -13,14 +13,14 @@ const { findImports } = require(`../utils/gen-mdx`)
 
 const contentDigest = val => createContentDigest(val)
 
-function unstable_shouldOnCreateNode({ node }, pluginOptions) {
+function shouldOnCreateNode({ node }, pluginOptions) {
   const options = defaultOptions(pluginOptions)
 
-  return _unstable_shouldOnCreateNode({ node }, options)
+  return _shouldOnCreateNode({ node }, options)
 }
 
 // eslint-disable-next-line camelcase
-function _unstable_shouldOnCreateNode({ node }, options) {
+function _shouldOnCreateNode({ node }, options) {
   // options check to stop transformation of the node
   if (options.shouldBlockNodeFromTransformation(node)) {
     return false
@@ -31,12 +31,12 @@ function _unstable_shouldOnCreateNode({ node }, options) {
     : options.mediaTypes.includes(node.internal.mediaType)
 }
 
-module.exports.unstable_shouldOnCreateNode = unstable_shouldOnCreateNode
+module.exports.shouldOnCreateNode = shouldOnCreateNode
 
 async function onCreateNode(api, pluginOptions) {
   const options = defaultOptions(pluginOptions)
 
-  if (!_unstable_shouldOnCreateNode({ node: api.node }, options)) {
+  if (!_shouldOnCreateNode({ node: api.node }, options)) {
     return
   }
 
diff --git a/gatsby-node.js b/gatsby-node.js
index 75b7c042a6d287b74c7584f942c67e626a6b2ac7..6bd725b8cd23abbb64de20db298bacff52670f22 100644
--- a/gatsby-node.js
+++ b/gatsby-node.js
@@ -6,7 +6,7 @@ const fs = require(`fs`)
 
 const {
   onCreateNode,
-  unstable_shouldOnCreateNode,
+  shouldOnCreateNode,
 } = require(`./gatsby/on-create-node`)
 
 /**
@@ -17,7 +17,7 @@ exports.createSchemaCustomization = require(`./gatsby/create-schema-customizatio
 /**
  * Check whether to create Mdx nodes from MDX files.
  */
-exports.unstable_shouldOnCreateNode = unstable_shouldOnCreateNode
+exports.shouldOnCreateNode = shouldOnCreateNode
 
 /**
  * Create Mdx nodes from MDX files.
