diff --git a/lib/jsdom/living/post-message.js b/lib/jsdom/living/post-message.js
index db16576db84d3aa45fe8469b4325447c945097d7..1fe3890dcf7cf22f48a62bd8e2bafa41f85cb1e1 100644
--- a/lib/jsdom/living/post-message.js
+++ b/lib/jsdom/living/post-message.js
@@ -6,7 +6,7 @@ const { isValidTargetOrigin } = require("../utils");
 const { fireAnEvent } = require("./helpers/events");
 
 module.exports = function (globalObject) {
-  return function (message, targetOrigin) {
+  return function (message, targetOrigin, ports) {
     if (arguments.length < 2) {
       throw new TypeError("'postMessage' requires 2 arguments: 'message' and 'targetOrigin'");
     }
@@ -33,7 +33,7 @@ module.exports = function (globalObject) {
     // TODO: event.ports
     // TODO: event.data - structured clone message - requires cloning DOM nodes
     setTimeout(() => {
-      fireAnEvent("message", this, MessageEvent, { data: message });
+      fireAnEvent("message", this, MessageEvent, { data: message, ports });
     }, 0);
   };
 };